node/src/WorkQueue.ts

```diff
@@ -47,6 +47,64 @@ export class WorkQueue {
     pipeline.lpush(this.mainQueueKey, item.id);
   }
 
+  /**
+   * Add an item to the work queue in a atomic way.
+   * This creates a pipeline and executes it on the database if the item id is not already in either the main queue or the procesing queue.
+   * 
+   * This method can be used only when database is locked
+   * 
+   * @param {Redis} db The Redis Connection.
+   * @param item The item that will be executed using the method addItemToPipeline.
+   */
+  async addAtomicItem(db: Redis, item: Item): Promise<boolean> {
+    const result = async (): Promise<boolean> => {
+      let transactionError = false;
+      do {
+        try {
+          await db.watch(this.mainQueueKey, this.processingKey);
+  
+          const pipeNew = db.pipeline();
+  
+          pipeNew.lpos(
+            this.processingKey,
+            item.id
+          );
+  
+          pipeNew.lpos(
+            this.mainQueueKey,
+            item.id
+          );
+  
+          const pipeResult = await pipeNew.exec();
+  
+          console.log(pipeResult);
+          if (!pipeResult || pipeResult[0][1] !== null || pipeResult[1][1] !== null) {
+            return false; // Item already exists in either mainQueue or processingKey
+          }
+  
+          const pipeEx = db.pipeline();
+          this.addItemToPipeline(pipeEx, item);
+          await pipeEx.exec();
```

Comment:
For `WATCH` to work, this has to wrap the command in `MULTI ... EXEC`. I don't think it does?